cscs-checks/microbenchmarks/gpu/memory_bandwidth/memory_bandwidth.py

```diff
@@ -99,21 +79,57 @@ def __init__(self):
                      'craype', 'external-resources'}
         self.maintainers = ['AJ', 'SK']
 
-    @rfm.run_before('compile')
-    def set_nvidia_sm_arch(self):
-        nvidia_sm = '60'
-        if self.current_system.name in ['arolla', 'tsa', 'ault']:
-            nvidia_sm = '70'
+    @rfm.run_after('setup')
+    def select_makefile(self):
+        cp = self.current_partition.fullname
+        if cp == 'ault:amdvega':
+            self.prebuild_cmds = ['cp makefile_memoryBandwidth.hip Makefile']
+        else:
+            self.prebuild_cmds = ['cp makefile_memoryBandwidth.cuda Makefile']
+
+    @rfm.run_after('setup')
+    def set_gpu_arch(self):
```

Comment:
I guess these functions will be moved to a "library test" as soon as we have have this functionality. How do you see that?